### PR TITLE
Add Github Enterprise Support.

### PR DESCRIPTION
**Pull request by hibooboo2**

### CODE REVIEW

**Comment by alena1108** [comment]
LGTM

File: code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/github/GithubClientEndpoints.java [hunk 1]
```diff
@@ -0,0 +1,8 @@
+package io.cattle.platform.iaas.api.auth.github;
+
+/**
+ * Created by wizardofmath on 5/26/15.
```
Comment (ibuildthecloud) [inline code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/github/GithubClientEndpoints.java]: Remove comment

File: code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/github/GithubConfigManager.java [hunk 2]
```diff
@@ -35,6 +35,7 @@
     private static final String CLIENT_SECRET = "clientSecret";
     private static final String ALLOWED_USERS = "allowedUsers";
     private static final String ALLOWED_ORGS = "allowedOrganizations";
+    private static final String HOST_NAME = "hostName";
```
Comment (ibuildthecloud) [inline code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/github/GithubConfigManager.java]: Can you make hostname lowercase.  I know host and name are two different words but geeks have unanimously agreed that hostname should be a word

Comment (ibuildthecloud) [inline code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/github/GithubConfigManager.java]: and HOSTNAME

File: code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/github/GithubUtils.java [hunk 3]
```diff
@@ -198,4 +204,40 @@ public String apply(String arg) {
         }
         return strings;
     }
+
+    public String getURL(GithubClientEndpoints val) {
+        String hostName;
+        String apiEndpoint;
+        if (GITHUB_HOSTNAME.get() == null){
```
Comment (ibuildthecloud) [inline code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/github/GithubUtils.java]: Do StringUtils.isBlank()

File: code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/github/GithubClientEndpoints.java (hunk 1 above)
Comment (hibooboo2) [inline code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/github/GithubClientEndpoints.java]: Oops lol darn intellij

**Pull request merged**